### PR TITLE
Fix/handle undefined value

### PR DESCRIPTION
**Pull request by shreeyash07**
- Handle undefined value in score card and country details
- design changes in country component

## Changes

* Detailed list or prose of changes
* Breaking changes
* Changes to configurations

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

### CODE REVIEW

File: app/components/ProgressBar/index.tsx [hunk 1]
```diff
@@ -15,6 +14,7 @@ export interface Props {
     value: number | undefined,
     subValue?: number,
     totalValue: number | undefined,
+    icon?: React.ReactNode
```
Comment (AdityaKhatri) [inline app/components/ProgressBar/index.tsx]:
```suggestion
    icon?: React.ReactNode;
```

File: app/views/Dashboard/Country/index.tsx [hunk 2]
```diff
@@ -157,21 +156,26 @@ function Country(props: Props) {
...
 
-    const outbreaks = countryResponse?.countryEmergencyProfile.map((item) => {
-        const colors = LINE_COLORS[Number(item.contextIndicatorValue) % LINE_COLORS.length];
-        return (
-            {
-                emergency: item.emergency,
-                fill: colors,
-            }
-        );
+    const outbreaks = unique(
```
Comment (AdityaKhatri) [inline app/views/Dashboard/Country/index.tsx]: use useMemo in all of the above functions

File: app/views/Dashboard/Country/index.tsx [hunk 3]
```diff
@@ -249,16 +255,19 @@ function Country(props: Props) {
...
+                                    renderer={ScoreCard}
+                                    rendererParams={readinessRendererParams}
+                                    data={scoreCardData}
+                                    keySelector={readinessKeySelector}
+                                    errored={false}
+                                    filtered={false}
+                                    pending={false}
+                                />
+                            )}
                     </ContainerCard>
```
Comment (AdityaKhatri) [inline app/views/Dashboard/Country/index.tsx]: fix indent

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/views/Dashboard/Country/index.tsx (hunk 2 above)
Comment (shreeyash07) [reply app/views/Dashboard/Country/index.tsx]: Done

File: app/views/Dashboard/Country/index.tsx (hunk 3 above)
Comment (shreeyash07) [reply app/views/Dashboard/Country/index.tsx]: Done

File: app/components/ProgressBar/index.tsx (hunk 1 above)
Comment (shreeyash07) [reply app/components/ProgressBar/index.tsx]: Done

File: app/components/ProgressBar/index.tsx [hunk 4]
```diff
@@ -15,6 +14,7 @@ export interface Props {
     value: number | undefined,
     subValue?: number,
     totalValue: number | undefined,
```
Comment (subinasr) [inline app/components/ProgressBar/index.tsx]:
```suggestion
    value: number | undefined;
    subValue?: number;
    totalValue: number | undefined;
```

File: app/views/Dashboard/CombinedIndicators/index.tsx [hunk 5]
```diff
@@ -45,6 +46,7 @@ function CombinedIndicators(props: Props) {
             subValue: data.subValue,
             totalValue: data.totalValue,
             color: data.color,
+            icon: <IoInformationCircleOutline />,
         }), [],
```
Comment (subinasr) [inline app/views/Dashboard/CombinedIndicators/index.tsx:50]:
```suggestion
        }), [barHeight],
```

**Review by priyesh777 — APPROVED** [review]

File: app/views/Dashboard/Country/index.tsx [hunk 6]
```diff
@@ -394,98 +416,108 @@ function Country(props: Props) {
...
+                                valueContainerClassName={styles.valueText}
+                                labelContainerClassName={styles.labelText}
+                                hideLabelColon
+                                label="Doctors and nurses per 999 people"
+                                value={(
+                                    <>
+                                        {countryResponse?.countryProfile.medicalStaff
+                                            ? (countryResponse
+                                                ?.countryProfile.medicalStaff)?.toFixed(0)
+                                            : 'N/A'}
```
Comment (subinasr) [inline app/views/Dashboard/Country/index.tsx]: If we've already checked this condition above, it'll never be false, which is why I think it's okay to just display the value without the 'N/A'

File: app/views/Dashboard/Country/index.tsx [hunk 7]
```diff
@@ -394,98 +416,108 @@ function Country(props: Props) {
...
+                        {stringency && (
+                            <TextOutput
+                                className={styles.countryTextOutput}
+                                valueContainerClassName={styles.valueText}
+                                labelContainerClassName={styles.labelText}
+                                hideLabelColon
+                                label="Stringency"
+                                value={(
+                                    <>
+                                        {stringency ? `${stringency}%` : 'N/A'}
```
Comment (subinasr) [inline app/views/Dashboard/Country/index.tsx]: Same thing. This condition will always return true.

File: app/views/Dashboard/Country/index.tsx [hunk 8]
```diff
@@ -501,21 +533,23 @@ function Country(props: Props) {
...
+                        {economicSupportIndex && (
+                            <TextOutput
+                                className={styles.countryTextOutput}
+                                valueContainerClassName={styles.valueText}
+                                labelContainerClassName={styles.labelText}
+                                hideLabelColon
+                                label="Economic support index"
+                                value={(
+                                    <>
+                                        {economicSupportIndex ? `${economicSupportIndex}%` : 'N/A'}
```
Comment (subinasr) [inline app/views/Dashboard/Country/index.tsx]: Same thing.

**Review by subinasr — CHANGES_REQUESTED** [review]

File: app/components/ScoreCard/index.tsx [hunk 9]
```diff
@@ -17,23 +17,25 @@ function ScoreCard(props: Props) {
...
+                <div className={styles.readinessTitle}>{title}</div>
+                <div className={_cs(
+                    styles.readinessValue,
+                )}
+                >
+                    {(value ? `${value}%` : 'N/A')}
+                </div>
             </div>
-        </div>
+        ) : null
```
Comment (AdityaKhatri) [inline app/components/ScoreCard/index.tsx]: if (!value) {
    return undefined;
}

return (
        <div className={_cs(
            className,
            styles.readinessContainer,
                {(value ? `${value}%` : 'N/A')}
            </div>
        </div>
);

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

**Comment by subinasr** [comment]
Also, please fix lint issues.

**Pull request merged**